zapcore/write_syncer.go

```diff
@@ -75,6 +77,117 @@ func (s *lockedWriteSyncer) Sync() error {
 	return err
 }
 
+type bufferWriterSyncer struct {
```

Comment:
should we move this whole type and methods into a separate file?

Reply:
This file already contains Lock syncer and MultiWrite syncer, I think it is a proper place for Buffer syncer :-)